examples/infill/infill.cpp

```diff
@@ -233,10 +233,22 @@ int main(int argc, char ** argv) {
     const bool add_bos = llama_vocab_type(model) == LLAMA_VOCAB_TYPE_SPM;
     LOG("add_bos: %d\n", add_bos);
 
+    bool suff_rm_leading_spc = params.escape;
+    if (suff_rm_leading_spc && params.input_suffix.find_first_of(" ") == 0 && params.input_suffix.size() > 1) {
+        params.input_suffix.erase(0, 1);
+        suff_rm_leading_spc = false;
+    }
```

Comment:
Hm, I could be missing something, but this whole section should not be needed.

If you keep everything else and just replace this block with:

```suggestion
    const bool suff_rm_leading_spc = true;
```

Can you find a case that does not match the Python version?

Reply:
yes, I can find cases where it misbehaves:
 with params.escape, if the suffix starts with a space, the tokenizer adds a space and THEN tokenizes the double space to '259' so we can't remove the surplus space anymore but would have to "translate" the '259' to '29871' to remove the space, same with more than one leading spaces...
without params.escape if the suffix starts with a space, we remove it so it does not reflect the original suffix

not sure what the python version does as I can't test it, but the above behaviour seems wrong

Reply:
Ok, that makes sense. Let's double check, just in case:
@kherud Could you please run the Python codellama with suffix with 2 leading spaces and see what tokens are produced for infill?

Reply:
There seems to be another issue too, in interactive mode the -e seems to only act on the initial input as the processing happens when the params are loaded in common.cpp:617
Is that a bug? seems to be a bug for infill and server infill, what about main, others?

Reply:
'<PRE> <SUF>': 1, 32007, 259, 32008, 32009
'<PRE> <SUF> ': 1, 32007, 259, 32008, 29871, 32009
'<PRE> <SUF>  ': 1, 32007, 259, 32008, 259, 32009
'<PRE> <SUF>   ': 1, 32007, 259, 32008, 1678, 32009
'<PRE> <SUF>    ': 1, 32007, 259, 32008, 268, 32009